evap/staff/views.py

```diff
@@ -2294,6 +2306,35 @@ class UserMergeSelectionView(FormView):
     form_class = UserMergeSelectionForm
     template_name = "staff_user_merge_selection.html"
 
+    def get_context_data(self, **kwargs) -> dict[str, Any]:
+        context = super().get_context_data(**kwargs)
+
+        class UserNameFromEmail(Func):
+            # django docs support our usage here:
+            # https://docs.djangoproject.com/en/5.0/ref/models/expressions/#func-expressions
+            # pylint: disable=abstract-method
+            template = "split_part(%(expressions)s, '@', 1)"
+
+        query = UserProfile.objects.annotate(username_part_of_email=UserNameFromEmail("email"))
+
+        users_with_merge_candidates = query.annotate(
+            merge_candidate_pk=query.filter(username_part_of_email=UserNameFromEmail(OuterRef("email")))
+            .exclude(email__gte=OuterRef("email"))
```

Comment:
While thinking about the "don't match user with themselves" and "we only want each pair once", I didn't put enough thought into _which_ pair we want to have here. In the issue, the requirement given is that
> the main user profile to be kept is the one with the higher ID.

But we should be able to incorporate this here by simply using
```suggestion
            .exclude(pk__gte=OuterRef("pk"))
```

Or, maybe it's easier to think about the set here with `filter` and `lt`, not sure about that.

Reply:
I usually find `filter` to be easier to think about